### DISCUSSION TITLE
How i can reload the datatable?

### DISCUSSION THREAD

**Discussion by joselrodriguezd** [Power-Components/livewire-powergrid#530] [Q&A] [answered]
Hello, first I congratulate you for your component, second how can I do to refresh the datatable when choosing another option in a select field.

I have a main page where I have 3 components.
1.- is the select field
2.- it is a table that shows some data depending on the selection.
3.- is the datatable that shows the data according to the selection in the select.

I have them connected by listeners, but in the backend it refreshes the information but in the front end it does not change the data in the datatable.
![2022-07-29_16h03_50](https://user-images.githubusercontent.com/48702587/181842847-5b0b9f7a-1d72-4964-8c13-3cf3a6c36c8b.png)
![2022-07-29_16h04_11](https://user-images.githubusercontent.com/48702587/181842859-e1dd165c-2cef-4e8f-a314-b7f51fcb1390.png)
![2022-07-29_16h04_28](https://user-images.githubusercontent.com/48702587/181842869-9a1acff1-01a9-46d6-b2e4-054f95ce21fc.png)
![2022-07-29_16h06_03](https://user-images.githubusercontent.com/48702587/181842980-cd337c96-28af-42e4-84bf-5619b62b04ae.png)
![2022-07-29_16h08_13](https://user-images.githubusercontent.com/48702587/181843222-0b34b3fa-9196-4577-9906-c5002f5b0f67.png)

**Comment by luanfreitasdev** [comment, marked as answer]
Hello! @joselrodriguezd,

First, the component that will receive the update is already listening through the `'pg:eventRefresh-default'` method. [(check in the listeners section).](https://livewire-powergrid.com/#/table/component-settings?id=event-listeners)

To update the component on top of the livewire dispatch:

```php
public function updated($name, $value)
{
   $this->emit('pg:eventRefresh-default'); // or emitTo(MyTable::class, 'pg:eventRefresh-default');
}
```

**Reply by joselrodriguezd** [reply]
Danke!!!

**Reply by joselrodriguezd** [reply]
Two design questions, how can I change the name of the Action column? and the other question, how can I make the excel file that is exported take the values in number format and not in text format?

**Reply by ashek1412** [reply]
Hi, I have two parameters which is used in data table query .  I cannot update the parameters in power grid table though its updated in parent component. 

![image](https://github.com/Power-Components/livewire-powergrid/assets/18645594/1687c105-bc6a-460a-b9f0-356bfa02b63f)
![image](https://github.com/Power-Components/livewire-powergrid/assets/18645594/669c68be-fc42-4358-8d2c-413df76683a6)
